src/app/client/src/app/modules/learn/components/course-consumption/assessment-player/assessment-player.component.ts

```diff
@@ -925,10 +927,10 @@ export class AssessmentPlayerComponent implements OnInit, OnDestroy, ComponentCa
   }
 
   onSelfAssessLastAttempt(event) {
-    if (_.get(event, 'data') === 'renderer:selfassess:lastattempt' || _.get(event, 'edata.isLastAttempt')) {
+    if (_.get(event, 'data') === 'renderer:selfassess:lastattempt' || _.get(event, 'edata.isLastAttempt' && !this.assessmenType)) {
```

Comment:
@sohailamjad12 If first condition true then if block will execute without checking assessmentType. Second condition after or operator is not correct.